### PR TITLE
Adapt stream breaking change and adding usage for Nullable.

### PR DESCRIPTION
**Pull request by katmsft**

### CODE REVIEW

**Comment by JinmingHu-MSFT** [comment]
@katmsft Please rebase on top of latest master branch and fix the CI first.

File: sdk/storage/inc/datalake/path_client.hpp [hunk 1]
```diff
@@ -32,54 +32,52 @@ namespace Azure { namespace Storage { namespace DataLake {
...
-    std::string ContentLanguage;
+    Azure::Core::Nullable<std::string> CacheControl;
+    Azure::Core::Nullable<std::string> ContentDisposition;
+    Azure::Core::Nullable<std::string> ContentEncoding;
+    Azure::Core::Nullable<std::string> ContentLanguage;
     int64_t ContentLength = int64_t();
-    int64_t RangeOffset = int64_t();
-    int64_t RangeLength = int64_t();
+    Azure::Core::Nullable<int64_t> RangeOffset = int64_t();
+    Azure::Core::Nullable<int64_t> RangeLength = int64_t();
```
Comment (JinmingHu-MSFT) [inline sdk/storage/inc/datalake/path_client.hpp:41]:
```suggestion
    Azure::Core::Nullable<int64_t> RangeOffset;
    Azure::Core::Nullable<int64_t> RangeLength;
```

File: sdk/storage/inc/datalake/protocol/datalake_rest_client.hpp [hunk 2]
```diff
@@ -164,9 +166,9 @@ namespace Azure { namespace Storage { namespace DataLake {
 
   struct SetAccessControlRecursiveResponse
   {
-    int32_t DirectoriesSuccessful = int32_t();
-    int32_t FilesSuccessful = int32_t();
-    int32_t FailureCount = int32_t();
+    int32_t DirectoriesSuccessful;
+    int32_t FilesSuccessful;
+    int32_t FailureCount;
```
Comment (JinmingHu-MSFT) [inline sdk/storage/inc/datalake/protocol/datalake_rest_client.hpp:171]: won't compiler give warnings for these uninitialized variables?

File: sdk/storage/src/datalake/path_client.cpp [hunk 3]
```diff
@@ -33,10 +33,11 @@ namespace Azure { namespace Storage { namespace DataLake {
       return std::string(begin, end);
     }
 
-    std::pair<int64_t, int64_t> GetOffsetLength(const std::string& rangeString)
+    std::pair<int64_t, Azure::Core::Nullable<int64_t>> GetOffsetLength(
```
Comment (JinmingHu-MSFT) [inline sdk/storage/src/datalake/path_client.cpp:36]: This is really ugly.

File: sdk/storage/src/datalake/path_client.cpp [hunk 4]
```diff
@@ -298,19 +300,18 @@ namespace Azure { namespace Storage { namespace DataLake {
       const std::string& destinationPath,
       const PathRenameOptions& options)
   {
-    std::string destinationFileSystem = options.DestinationFileSystem;
-    if (destinationFileSystem.empty())
+    Azure::Core::Nullable<std::string> destinationFileSystem = options.DestinationFileSystem;
```
Comment (JinmingHu-MSFT) [inline sdk/storage/src/datalake/path_client.cpp:303]: What if `options.DestinationFileSystem` isn't null but an empty string?

File: sdk/storage/inc/datalake/datalake_utilities.hpp [hunk 5]
```diff
@@ -12,9 +13,10 @@
 namespace Azure { namespace Storage { namespace DataLake { namespace Details {
   UrlBuilder GetBlobUriFromDfsUri(const UrlBuilder& dfsUri);
```
Comment (JinmingHu-MSFT) [inline sdk/storage/inc/datalake/datalake_utilities.hpp:15]: These two functions will finally be removed, right?

File: sdk/storage/inc/datalake/protocol/datalake_rest_client.hpp [hunk 6]
```diff
@@ -594,16 +596,16 @@ namespace Azure { namespace Storage { namespace DataLake {
...
+    Azure::Core::Nullable<std::string> ContentLanguage;
     int64_t ContentLength = int64_t();
-    std::string ContentRange;
+    Azure::Core::Nullable<std::string> ContentRange;
     std::string ContentType;
     std::string ContentMD5;
-    std::string Properties;
-    std::string XMsContinuation;
+    Azure::Core::Nullable<std::string> Properties;
+    Azure::Core::Nullable<std::string> XMsContinuation;
```
Comment (JinmingHu-MSFT) [inline sdk/storage/inc/datalake/protocol/datalake_rest_client.hpp:608]:
```suggestion
    Azure::Core::Nullable<std::string> Continuation;
```

File: sdk/storage/src/datalake/file_system_client.cpp [hunk 7]
```diff
@@ -140,7 +140,7 @@ namespace Azure { namespace Storage { namespace DataLake {
         std::move(result.LastModified),
         std::move(result.RequestId),
         std::move(result.Version),
-        Details::DeserializeMetadata(result.Properties),
+        Details::DeserializeMetadata(result.Properties).GetValue(),
```
Comment (JinmingHu-MSFT) [inline sdk/storage/src/datalake/file_system_client.cpp:143]: What if `result.Properties` is null? `.GetValue()` will crash the program.

File: sdk/storage/src/datalake/path_client.cpp [hunk 8]
```diff
@@ -437,28 +443,36 @@ namespace Azure { namespace Storage { namespace DataLake {
...
-                            std::move(result.LeaseDuration),
-                            std::move(result.LeaseState),
-                            std::move(result.LeaseStatus),
-                            std::move(result.XMsContentMd5),
-                            Details::DeserializeMetadata(result.Properties)};
+    Azure::Core::Nullable<int64_t> RangeOffset;
+    Azure::Core::Nullable<int64_t> RangeLength;
+    if (result.ContentRange.HasValue())
+    {
+      auto range = GetOffsetLength(result.ContentRange.GetValue());
```
Comment (JinmingHu-MSFT) [inline sdk/storage/src/datalake/path_client.cpp:450]: When will `range.second` be null?

**Comment by katmsft** [comment]
No longer valid, will push on top of https://github.com/Azure/azure-sdk-for-cpp/pull/236 instead.

File: sdk/storage/inc/datalake/protocol/datalake_rest_client.hpp (hunk 2 above)
Comment (katmsft) [reply sdk/storage/inc/datalake/protocol/datalake_rest_client.hpp:171]: I don't see any warnings from my side. Also, CI passes.

File: sdk/storage/src/datalake/path_client.cpp (hunk 4 above)
Comment (katmsft) [reply sdk/storage/src/datalake/path_client.cpp:303]: Changed to honor the current file system if the destination is an empty string. However, this does not look perfect for me. Setting destination file system to be empty should be undefined behavior.

File: sdk/storage/inc/datalake/datalake_utilities.hpp (hunk 5 above)
Comment (katmsft) [reply sdk/storage/inc/datalake/datalake_utilities.hpp:15]: `GetBlobUriFromDfsUri` will be removed only after full interop is implemented.

File: sdk/storage/src/datalake/path_client.cpp (hunk 8 above)
Comment (katmsft) [reply sdk/storage/src/datalake/path_client.cpp:450]: When `result.ContentRange` is null for the response and when customer specifies `bytes=<somevalue>-` for requests.

File: sdk/storage/inc/datalake/datalake_utilities.hpp (hunk 5 above)
Comment (JinmingHu-MSFT) [reply sdk/storage/inc/datalake/datalake_utilities.hpp:15]: I meant `DeserializeMetadata` and `SerializeMetadata`

File: sdk/storage/src/datalake/path_client.cpp (hunk 8 above)
Comment (JinmingHu-MSFT) [reply sdk/storage/src/datalake/path_client.cpp:450]: I doubt that

> Host: adlsgen2account.dfs.core.windows.net
> User-Agent: curl/7.54.0
> Accept: */*
> Range: bytes=1-
> 
< HTTP/1.1 206 Partial Content
< Content-Length: 11
< Content-Type: application/octet-stream
< Content-Range: bytes 1-11/12
< Last-Modified: Thu, 02 Jul 2020 15:37:28 GMT
< Accept-Ranges: bytes
< ETag: "0x8D81E9DD3983864"
< Server: Windows-Azure-HDFS/1.0 Microsoft-HTTPAPI/2.0
< x-ms-properties: 
< x-ms-resource-type: file
< x-ms-lease-state: available
< x-ms-lease-status: unlocked
< x-ms-server-encrypted: false
< x-ms-content-md5: b1kCrCNwJL3QwXbLkwY9xA==
< x-ms-request-id: 60e167a9-d01f-0022-0e86-50aade000000
< x-ms-version: 2019-10-10
< Date: Thu, 02 Jul 2020 15:38:26 GMT
<

File: sdk/storage/inc/datalake/protocol/datalake_rest_client.hpp (hunk 2 above)
Comment (JinmingHu-MSFT) [reply sdk/storage/inc/datalake/protocol/datalake_rest_client.hpp:171]: fine, if compilers say it's okay, then i'm okay.

**Pull request closed without merging**